lib/blpapi-wrapper.ts

```diff
@@ -324,7 +334,11 @@ export class Session extends events.EventEmitter {
                 throw new Error('Unknown request type ' + requestName);
             }
             var responseEventName = REQUEST_TO_RESPONSE_MAP[requestName];
-            this.session.request(uri, requestName, request, correlatorId);
+            if (isAuthRequest) {
```

Comment:
Well, if this was two separate functions, each would be 20 lines, with exactly one line difference (`openService` promise + listening logic). I feel like this sharing is not great but it's better for future proofing to any other changes we might make here.